spec/models/mixins/relationship_mixin_spec.rb

```diff
@@ -216,11 +216,20 @@ def self.name
           has_ancestry
           self.table_name = "services"
           include RelationshipMixin
+          self.default_relationship_type = "ems_metadata"
           self.skip_relationships += ["custom"]
         end
       end
 
+      # ems_metadata:
+      #   1
+      #  2
+      # custom:
+      #   2
+      #  1
+      # 3 4
       before do
+        service2.update!(:parent => service1)
         service1.with_relationship_type("custom") { service1.update!(:parent => service2) }
```

Comment:
can i instead of update! for this and the next three lines use `service1.with_relationship_type("custom") { service1.add_parent(service2) }` ? That makes these tests with this code more green?